scripts/aem.js

```diff
@@ -419,40 +419,62 @@ function decorateLayouts(element) {
     // only creates a layout container for valid layout opening and closing tag pairs
     if (layoutClosingTag) {
       const layoutContainer = document.createElement('div');
-      layoutContainer.classList.add('grid-container');
-
-      // layout modifiers
-      if (layoutOpeningTag.textContent.includes('spacious'))
-        layoutContainer.classList.add('grid-container--large-gap');
-      if (layoutOpeningTag.textContent.includes('scrolling'))
-        layoutContainer.classList.add('grid-container--with-scroll');
-
-      elementsToContain.forEach((e, idx) => {
-        const gridItemContainer = document.createElement('div');
-        gridItemContainer.classList.add('grid-item');
-        gridItemContainer.append(e);
-
-        // apply two-up layout
-        if (layoutOpeningTag.textContent.endsWith('two-up-'))
-          gridItemContainer.classList.add('grid-item--50');
-
-        // apply three-up layout
-        if (layoutOpeningTag.textContent.endsWith('three-up-'))
-          gridItemContainer.classList.add('grid-item--30');
-
-        // apply 70/30 layout
-        if (layoutOpeningTag.textContent.endsWith('70-30-')) {
-          (idx % 2 === 0)
-            ? gridItemContainer.classList.add('grid-item--66')
-            : gridItemContainer.classList.add('grid-item--33');
-        };
+      // "full" layout name doesn't need grid-container and grid-items; it's just used with "hide-*" classes.
+      const skipGridContainers = layoutOpeningTag.textContent.endsWith('full-');
+
+      if (!skipGridContainers) {
+        layoutContainer.classList.add('grid-container');
 
-        // apply four-up layout
-        if (layoutOpeningTag.textContent.endsWith('four-up-'))
-          gridItemContainer.classList.add('grid-item--25');
+        // Layout modifiers - grid spacing and scrolling
+        if (layoutOpeningTag.textContent.includes('spacious'))
+          layoutContainer.classList.add('grid-container--large-gap');
+        if (layoutOpeningTag.textContent.includes('scrolling'))
+          layoutContainer.classList.add('grid-container--with-scroll');
+      }
 
-        layoutContainer.append(gridItemContainer);
+      // Layout modifiers - hiding at breakpoints
+      const hideModifiers = [
```

Comment:
this i like a lot

Reply:
can we break this out to a few functions? these names are probably bad

`applyBehaviorModifiers()`
`applyHideModifiers()`
`applyGridWrappers()`